pkg/controller/controller.go

```diff
@@ -148,6 +148,12 @@ func (c *Controller) Run() error {
 			pt.start()
 
 			clus := event.Object
+
+			if clus.Status.IsFailed() {
```

Comment:
for safety reasons, can we still check if cluster is inside the clusters map?

like change line 173 to 

```

c, ok := c.clusters[clus.Metadata.Name]
if !ok {
   ...
}
```

```